File: PyProject/geeksforgeeks/bubble_sort.py
# -*- coding:utf-8 -*-
# author: youdi

from typing import List


def bubble_sort(nums: List[int]) -> List[int]:
    nums_length = len(nums)
    for i in range(1, nums_length):
        for j in range(0, nums_length - i):
            if nums[j] > nums[j + 1]:
                nums[j], nums[j + 1] = nums[j + 1], nums[j]
    return nums


if __name__ == '__main__':
    A = [64, 25, 12, 22, 11]
    B = bubble_sort(A)
    print(B)

File: GoProject/leetcode/sort/history/bubble_sort.go
package main

func bubbleSort(nums []int) {
	l := len(nums)

	swap := true

	for i := 0; i < l-1; i++ {
		if !swap {
			break
		}
		swap = false
		for j := 0; j < l-1-i; j++ {
			if nums[j] > nums[j+1] {
				nums[j], nums[j+1] = nums[j+1], nums[j]
				swap = true
			}
		}
	}
}

func main() {
	a := []int{5, 2, 3, 1}
	bubbleSort(a)
	for _, v := range a {
		println(v)
	}
}
